src/gui/src/settings/options-window.cpp

```diff
@@ -342,6 +343,151 @@ OptionsWindow::OptionsWindow(Profile *profile, ThemeLoader *themeLoader, QWidget
+					ui->lineButtonSaveNQuit->setText(button->states[0].text.isEmpty() ? Ui::DefaultSaveNQuitStateSave.text : button->states[0].text);
+					ui->lineButtonSaveNQuit->setCursorPosition(0);
+					break;
+				case Ui::IsButtonOpen :
+					//log("Open");
+					checker = ui->checkButtonOpen;
+					positionSpinner = ui->spinButtonOpenPosition;
+					widthSpinner = ui->spinButtonOpenWidth;
+					ui->lineButtonOpen->setText(button->states[0].text.isEmpty() ? Ui::DefaultOpenState.text : button->states[0].text);
+					ui->lineButtonOpen->setCursorPosition(0);
+					break;
+				case Ui::IsButtonSave | Ui::IsFavoriteButton :
+					//log("SaveFav");
+					checker = ui->checkButtonSaveFav;
+					positionSpinner = ui->spinButtonSaveFavPosition;
+					widthSpinner = ui->spinButtonSaveFavWidth;
+					ui->lineButtonSaveFav->setText(button->states[0].text.isEmpty() ? Ui::DefaultSaveFavStateSave.text : button->states[0].text);
+					ui->lineButtonSaveFav->setCursorPosition(0);
+					break;
+				case Ui::IsButtonSaveNQuit | Ui::IsFavoriteButton :
+					//log("SaveNQuitFav");
+					checker = ui->checkButtonSaveNQuitFav;
+					positionSpinner = ui->spinButtonSaveNQuitFavPosition;
+					widthSpinner = ui->spinButtonSaveNQuitFavWidth;
+					ui->lineButtonSaveNQuitFav->setText(button->states[0].text.isEmpty() ? Ui::DefaultSaveNQuitFavStateSave.text : button->states[0].text);
+					ui->lineButtonSaveNQuitFav->setCursorPosition(0);
+					break;
+				case Ui::IsButtonOpen | Ui::IsFavoriteButton :
+					//log("OpenFav");
+					checker = ui->checkButtonOpenFav;
+					positionSpinner = ui->spinButtonOpenFavPosition;
+					widthSpinner = ui->spinButtonOpenFavWidth;
+					ui->lineButtonOpenFav->setText(button->states[0].text.isEmpty() ? Ui::DefaultOpenFavState.text : button->states[0].text);
+					ui->lineButtonOpenFav->setCursorPosition(0);
+					break;
+				default :
+					log("OptionsWindow found an unknown button type.");
+					continue;
+			}
+			log(button->name.c_str());
+			//log(std::to_string(button->type).c_str());
+			checker->setCheckState(button->isEnabled ? button->isInDrawer ? Qt::PartiallyChecked : Qt::Checked : Qt::Unchecked);
+			positionSpinner->setValue(button->position);
+			widthSpinner->setValue(button->relativeWidth);
+		}
+		settings->endGroup();
+		log("---Reading Zoom/Buttons---");
+
+		QList<QGroupBox*> buttonGroups = ui->pageInterfaceImageWindowButtons->findChildren<QGroupBox *>();
+		//QList<std::pair<QCheckBox*, QSpinBox*>> csPairs;
+		for (unsigned short i = 0; i < buttonGroups.count(); i++) {
+			// Note that this will break if multiple checkboxes or spinners are in any group.
+			csPairs.append(std::make_pair(
+				buttonGroups.at(i)->findChild<QCheckBox *>(),
+				buttonGroups.at(i)->findChild<QSpinBox *>()
+			));
+		}
+		//checkAllSpinners(&csPairs);
+		checkAllSpinners();
+		QObject::connect(ui->spinButtonPrevPosition,		static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this, &OptionsWindow::checkSpinners);
```

Comment:
Since you have the list in csPairs, you could iterate  on them?
You can do all those connections in the UI file directly, which would be even better.

Reply:
Good ideas! I'll do that.

Reply:
I pushed an update that does everything there in one loop. Really not sure why I didn't do that to begin with. I also have a version that makes the connections in the .ui file, but if we use that then we will have to set blockSignals(true) and then blockSignals(false) for each "checker" and "spinner". I'm afraid that would be less efficient and make the code a little messier. Do you know?

Reply:
Doing it code-side allows for more flexibility so you can keep the loop instead of the UI change.